botorch/exceptions/warnings.py

```diff
@@ -49,3 +49,10 @@ class BotorchTensorDimensionWarning(BotorchWarning):
     r"""Warning raised when a tensor possibly violates a botorch convention."""
 
     pass
+
+
+class ProximalWarning(BotorchWarning):
```

Comment:
I think this is a bit too bespoke a warning - can you just use a `BoTorchWarning` here?